pennylane/tape/tape.py

```diff
@@ -554,12 +581,12 @@ def _update(self):
         self._graph = None
         self._specs = None
         self._depth = None
-        self._update_circuit_info()
-        self._update_par_info()
-        self._update_trainable_params()
-        self._update_observables()
-        self._update_batch_size()
-        self._update_output_dim()
+        self._update_circuit_info()  # Updates wires, num_wires, is_sampled, all_sampled; O(ops+obs)
+        self._update_par_info()  # Updates the _par_info dictionary; O(ops+obs)
+        self._update_trainable_params()  # Updates the _trainable_params; O(1)
+        self._update_observables()  # Updates _obs_sharing_wires and _obs_sharing_wires_id
+        self._update_batch_size()  # Updates _batch_size; O(ops)
+        self._update_output_dim()  # Updates _output_dim; O(obs)
```

Comment:
How about we update the order of these methods here so that it's the same order the methods are defined in?
```suggestion
        self._update_circuit_info()  # Updates wires, num_wires, is_sampled, all_sampled; O(ops+obs)
        self._update_batch_size()  # Updates _batch_size; O(ops)
        self._update_output_dim()  # Updates _output_dim; O(obs)
        self._update_observables()  # Updates _obs_sharing_wires and _obs_sharing_wires_id
        self._update_par_info()  # Updates the _par_info dictionary; O(ops+obs)
        self._update_trainable_params()  # Updates the _trainable_params; O(1)
```

Reply:
And document or group any potential interactions.

Reply:
Done.